iroh-api/src/p2p.rs

```diff
@@ -56,9 +85,69 @@ impl P2p {
             .await
             .map_err(|e| map_service_error("p2p", e))
     }
+
+    /// Get a stream of [`NetworkEvent`].
+    pub async fn network_events(&self) -> Result<BoxStream<'static, Result<NetworkEvent>>> {
+        let stream = self
+            .client
+            .network_events()
+            .await
+            .map_err(|e| map_service_error("p2p", e))?;
+        Ok(stream.boxed())
+    }
+
+    /// Subscribe to a Gossipsub Topic
+    ///
+    /// Gossipsub is a pub/sub protocol. This will subscribe you to a Gossipsub
+    /// topic. All Gossipsub messages for topics that you are subscribed to can
+    /// be read off the the `NetworkEvent` stream, as
+    /// `NetworkEvent::Gossipsub(GossipsubEvent::Message)`. You can access this
+    /// stream using `P2p::network_events`.
```

Comment:
This makes me wonder if the API would be better if it had a separate stream for just the gossipsub topic you request.  We could even return that stream right here.